### DISCUSSION TITLE
bracket syntax vs. dot syntax (backward-forward) compatibility

### DISCUSSION THREAD

**Discussion by dwiyatci** [react-hook-form/react-hook-form#5163] [Q&A] [answered]
Hi -- just a quick question here. Since I'm still juggling between v6 and v7 now, I want to double-check again:

1. Can the bracket syntax, e.g. `test[2].test`, still be used in v7?
2. Conversely, can the dot syntax, e.g. `test.2.test` be used in v6?

I've read the [migration guide](https://react-hook-form.com/migrate-v6-to-v7/), but I just wanted to have a more firm clarification whether it's not supported or just deprecated.

Thank you :)

**Comment by dwiyatci** [comment, marked as answer]
Ok, after some more extensive tests, I think they still work interchangeably in v6 an v7, at least for `[${index}]` and `.${index}`. @bluebill1049 Could you give a quick "Yes/No" confirmation here? 😬🙏

**Reply by bluebill1049** [reply]
please stay with dot syntax in v7.

**Reply by dwiyatci** [reply]
Yes, that's what I prefer now. Even in my v6 code, I switch to use dot syntax only to ease the migration later on. Thanks for the confirmation 🙇

**Reply by dwiyatci** [reply]
Note-to-self (or to dear other readers): ca. 3 months later after my last comment above, I found out that **there are common-cases where dot syntax makes some behaviour break for our base code that's still using v6**. ⚠️ 😮‍💨

So I guess it's safer to stay with the bracket syntax in v6.
